core/policy/policy_repository.go

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright (C) 2019 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package policy
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/mysteriumnetwork/node/logconfig/httptrace"
+	"github.com/mysteriumnetwork/node/market"
+	"github.com/mysteriumnetwork/node/requests"
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog/log"
+)
+
+type policyMetadata struct {
+	policy market.AccessPolicy
+	eTag   string
+	rules  market.AccessPolicyRuleSet
+}
+
+// PolicyRepository represents async policy fetcher from TrustOracle
+type PolicyRepository struct {
+	client *requests.HTTPClient
+
+	policyURL  string
+	policyLock sync.Mutex
+	policyList []policyMetadata
+
+	fetchInterval time.Duration
+	fetchShutdown chan struct{}
+}
+
+// NewPolicyRepository create instance of policy repository
+func NewPolicyRepository(client *requests.HTTPClient, policyURL string, interval time.Duration) *PolicyRepository {
+	return &PolicyRepository{
+		client:        client,
+		policyURL:     policyURL,
+		policyList:    make([]policyMetadata, 0),
+		fetchInterval: interval,
+		fetchShutdown: make(chan struct{}),
+	}
+}
+
+// Start begins fetching policies to repository
+func (pr *PolicyRepository) Start() {
+	go pr.fetchLoop()
+}
+
+// Stop ends fetching policies to repository
+func (pr *PolicyRepository) Stop() {
+	pr.fetchShutdown <- struct{}{}
+}
+
+// Policy converts given value to valid policy rule
+func (pr *PolicyRepository) Policy(policyID string) market.AccessPolicy {
+	policyURL := pr.policyURL
+	if !strings.HasSuffix(policyURL, "/") {
```

Comment:
It can be 3rd party service, just follows this contract:

1. While starting service, provide Policy Oracle via:
  `--access-policy.address=https://testnet-trust.mysterium.network/api/v1/access-policies/` 
  `--access-policy.address:https://testnet-trust.mysterium.network/api/v1/access-policies`
2. Provider policy IDs via:
  `--access-policy.list= ipinfo,wikipedia`

So yes, provided custom URL still have to match certain contract

Reply:
But you don't need to add `/` at the end of the user URL. 
If you need it for URL `https://testnet-trust.mysterium.network/api/v1/access-policies/` you passing it through the command argument. 

If I need to pass the URL `https://myown-trust.com/policies.json`, it should not become `https://myown-trust.com/policies.json/`.